### PR TITLE
Fix f18 install method

### PR DESCRIPTION
**Pull request by naromero77**
Resolve https://github.com/spack/spack/issues/13118

@alalazo @certik

### CODE REVIEW

**Comment by certik** [comment]
Nice, the built-in install method didn't use to work, but it looks like it does now. I don't have time right now to check this if this actually works --- does this install the `f18` binary correctly?

File: var/spack/repos/builtin/packages/f18/package.py [hunk 1]
```diff
@@ -17,7 +17,3 @@ class F18(CMakePackage):
     version('develop', branch='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/f18/package.py]: You can actually rename this version to `master` now that #1983 has been merged.

**Comment by naromero77** [comment]
> Nice, the built-in install method didn't use to work, but it looks like it does now. I don't have time right now to check this if this actually works --- does this install the `f18` binary correctly?

Yes, it builds correctly and installs. I tested it on Ubuntu 18.04 TLS with GCC 8.3.x

File: var/spack/repos/builtin/packages/f18/package.py (hunk 1 above)
Comment (naromero77) [reply var/spack/repos/builtin/packages/f18/package.py]: @certik @adamjstewart Sorry, I am not familiar with this package. I get the jist of the `Version` pull request, but did not follow the details.

You are saying to change the line to?

`version('master')` or

`version('master', branch='master')`

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/f18/package.py]: The latter

**Comment by chissg** [comment]
Is there any remaining reason for the version prescription on the LLVM dependency? There's no such prescription I could find in the [current documentation](https://github.com/flang-compiler/f18/blob/master/README.md), but it does mention having been tested against Clang 8.0 and 9.0.

**Comment by naromero77** [comment]
> Is there any remaining reason for the version prescription on the LLVM dependency? There's no such prescription I could find in the [current documentation](https://github.com/flang-compiler/f18/blob/master/README.md), but it does mention having been tested against Clang 8.0 and 9.0.

@chissg 

This PR request is about fixing the install method. 

I am not sure about the LLVM dependency as I am not very familiar with this package. However, if you tell me what it should be, I can create another PR and get those updates in. I can also go off the f18 docs online. We should get a couple of variants in there was well (release, debug, etc.).

**Comment by adamjstewart** [comment]
> We should get a couple of variants in there was well (release, debug, etc.).

These are already added by the `CMakePackage` base class in the form of the `build_type` variant.

**Comment by naromero77** [comment]
Right

Also, if someone knows what versions are needed beside master, please let me know.

**Comment by naromero77** [comment]
@adamjstewart Can you merge in?

**Comment by chissg** [comment]
@naromero77 Currently there are no tags, to my knowledge. At some point within the next few months this will be renamed Flang and should replace the existing Flang recipe.

I would remove the version specification from the `depends_on` directive. When conflicts are identified or (specified in the documentation) they can be added to the recipe.

**Pull request merged**